test/unit/compose/service.spec.ts

```diff
@@ -658,6 +659,66 @@ describe('compose/service: unit tests', () => {
 			expect(svc1.isEqualConfig(svc2, {})).to.be.true;
 			expect(svc2.isEqualConfig(svc1, {})).to.be.true;
 		});
+
+		it('should redact environment variables from debug logs when configs differ', async () => {
+			const svc1 = await Service.fromComposeObject(
+				{
+					appId: 1,
+					serviceId: 1,
+					serviceName: 'test',
+					environment: {
+						SECRET_KEY: 'super-secret-value',
+						API_TOKEN: 'sensitive-token',
+					},
+				},
+				{ appName: 'test' } as any,
+			);
+
+			const svc2 = await Service.fromComposeObject(
+				{
+					appId: 1,
+					serviceId: 1,
+					serviceName: 'test',
+					environment: {
+						SECRET_KEY: 'different-secret',
+						API_TOKEN: 'different-token',
+						NEW_VAR: 'new-value',
+					},
+				},
+				{ appName: 'test' } as any,
+			);
+
+			// Capture debug logs - log.debug may already be a spy from test setup
+			const debugCalls: string[] = [];
+			const originalFn = log.debug;
+			log.debug = (...args: any[]) => {
+				debugCalls.push(args.join(' '));
+				return originalFn.apply(log, args);
+			};
```

Comment:
You should be able to query the spy calls directly with `(log.debug as SinonSpy)`. See Spy API under https://sinonjs.org/releases/latest/spies/